event-driven-architecture/src/main/java/com/iluwatar/eda/framework/DynamicRouter.java

```diff
@@ -0,0 +1,13 @@
+package com.iluwatar.eda.framework;
+
+/**
+ * A {@link DynamicRouter} is responsible for selecting the proper path of a {@link Message}
+ * Messages can be associated to Channels through the registerChannel method and dispatched by
+ * calling the dispatch method.
+ */
+public interface DynamicRouter<E extends Message> {
```

Comment:
Should we have to name this interface as `EventDispatcher` and rename existing dispatcher to be `ContentTypeBasedDispatcher`. Also what can be other types of dispatchers? I am concerned about the need to this interface if there isn't any other use case. We should have one more implementation as well which utilizes the interface.